com.ibm.streamsx.topology/opt/python/packages/streamsx/topology/consistent.py

```diff
@@ -0,0 +1,159 @@
+
+    .. seealso:: :py:meth:`~streamsx.topology.topology.Stream.set_consistent`
+
+    """
+
+    class Trigger(Enum):
+        """
+        Defines how the drain-checkpoint cycle of a consistent region is triggered.
+        """
+        OPERATOR_DRIVEN = 1
+        """
+        Region is triggered by the start operator.
+        """
+        PERIODIC = 2
+        """
+        Region is triggered periodically.
+        """
+
+    def __init__(self, trigger=None, period=None, drainTimeout=None, resetTimeout=None, maxConsecutiveAttempts=None):
+
+        # period cannot be specified for OPERATOR_DRIVEN
+        # (This can only happen if someone creates an instance 
+        # directly instead of using the periodic and operator_driven
+        # methods.
+        if trigger == ConsistentRegionConfig.Trigger.OPERATOR_DRIVEN and period is not None:
+            raise ValueError("period does not apply to an operator driven consistent region")
+
+        if trigger == ConsistentRegionConfig.Trigger.PERIODIC:
+            if period is None:
+                raise ValueError("period must be specified for a consistent region with periodic trigger.")
+            elif isinstance(period, timedelta):
+                if period.total_seconds() <= 0.0:
+                    raise ValueError("period must be greater than zero.")
+            elif float(period) <= 0.0:
+                raise ValueError("period must be greater than zero.")
+
+
+        # drainTimeout and resetTimeout must be timedelta values, or must be castable to
+        # float, and both must be greater than 0.
+        if drainTimeout is None:
+            pass
+        elif isinstance(drainTimeout, timedelta):
+            if drainTimeout.total_seconds() <= 0.0:
+                raise ValueError("drain timeout value must be greater than zero.")
+        elif float(drainTimeout) <= 0.0:
+            raise ValueError("drain timeout value must be greater than zero.")
+
+
+        if resetTimeout is None:
+            pass
+        elif isinstance(resetTimeout, timedelta):
+            if resetTimeout.total_seconds() <= 0.0:
+                raise ValueError("reset timeout value must be greater than zero.")
+        elif float(resetTimeout) <= 0.0:
+            raise ValueError("reset timeout value must be greater than zero.")
+
+
+        # maxConsecutiveAttempts must be 1-0x7FFFFFFF.  It also must be an int.
+        if maxConsecutiveAttempts is None:
+            pass
+        elif int(maxConsecutiveAttempts) < 1 or int(maxConsecutiveAttempts) > 0x7FFFFFFF:
+            raise ValueError("maxConsecutiveAttempts must be between 1 and " + str(0x7FFFFFFF) + ", inclusive.")
+        elif not float(maxConsecutiveAttempts).is_integer():
+            raise ValueError("maxConsecutiveAttempts must be an integer value.")
+
+        self.trigger = trigger
+        self.period = period
+        self.drainTimeout = drainTimeout if drainTimeout is not None else 180
```

Comment:
Could just have the initial method default to 180 instead of None.

Reply:
Changed.